packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -665,6 +665,14 @@ export class SchemaFactory<
 		return new FieldSchema(FieldKind.Optional, t);
 	}
 
+	/**
+	 * Make a field of type identifier instead of the default which is required.
+	 */
+	public identifier<const T extends TreeNodeSchema>(t: T): FieldSchema<FieldKind.Identifier, T> {
```

Comment:
I know that you and Craig said we can't hardcode the underlying IdentifierFlexField to only contain strings, but can we make _this_ function only accept strings? We need to avoid making the user pass the type of the identifier in when they declare their schema. We want this:

`sf.object("o", { id: sf.identifier() })`

not this:

`sf.object("o", { id: sf.identifier(sf.string) })`

Reply:
```ts
	public get identifier(): FieldSchema<
		FieldKind.Identifier,
		TreeNodeSchema<"com.fluidframework.leaf.string", NodeKind.Leaf, string, string>
	> {
		return new FieldSchema(FieldKind.Identifier, this.string);
	}
```

seems to work

Reply:
Updated to hardcode use of `this.string`